Update state version and workspace API docs (#379)
hashicorp/terraform-docs-common: website/docs/cloud-docs/workspaces/state.mdx

In local runs (available for workspaces whose execution mode setting is set to "local"), you can use a workspace's state by configuring the [CLI integration](/terraform/cli/cloud) and authenticating with a user token that has permission to read and write state versions for the relevant workspace. When using a Terraform configuration that references outputs from another workspace, the authentication token must also have permission to read state outputs for that workspace. ([More about permissions.](/terraform/cloud-docs/users-teams-organizations/permissions))

> During a Terraform Cloud run, Terraform incrementally creates provisional state versions and marks them as finalized once it uploads the state content. When a workspace is unlocked will Terraform Cloud select the latest finalized state version as the current state version for the workspace. If the latest state version is not finalized when a workspace is unlocked, the workspace may need to be force unlocked. In the event of a force unlock, Terraform Cloud will ensure to select the latest finalized state version. When Terraform Cloud selects the current state version, all other provisional state versions are discarded.

Reviewer: Suggestion to split this up for readability and clarity:
```suggestion
During a Terraform Cloud run, Terraform incrementally creates provisional state versions and marks them as finalized once it uploads the state content.  When a workspace is unlocked, Terraform Cloud selects the latest finalized state version as the current state version for that workspace. 

If Terraform Cloud has not finalized the latest state version when a workspace is unlocked, the workspace may need to be force unlocked. In the event of a force unlock, Terraform Cloud selects the latest finalized state version. When Terraform Cloud selects the current state version, it discards all other provisional state versions.
```
